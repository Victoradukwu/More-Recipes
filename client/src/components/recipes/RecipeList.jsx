import React from 'react';
import PropTypes from 'prop-types';
import UserRecipe from './UserRecipe';

const RecipeList = ({ userRecipes }) => {
  return (
    <table className="table">
      <thead>
        <tr>
          <th>Recipe Name</th>
          <th>Upvotes</th>
          <th>Downvotes</th>
          <th>Favorites</th>
          <th>views</th>
          <th>Actions</th>        
        </tr>
      </thead>
      <tbody>
        { userRecipes.map(userRecipe =>
          <UserRecipe key={userRecipe.Id} userRecipe={userRecipe} />) }

      </tbody>
    </table>
  );
};
RecipeList.propTypes = {
  userRecipes: PropTypes.array.isRequired
};

export default RecipeList;
