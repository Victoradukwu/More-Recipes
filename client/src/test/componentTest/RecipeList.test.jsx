import React from 'react';
import { shallow } from 'enzyme';
import RecipeList from '../../components/recipes/RecipeList';

describe('RecipeList', () => {
  const props = {
    userRecipes: [{ id: 1 }],
    deleteRecipe: jest.fn()
  };
  it('matches snapshot', () => {
    const wrapper = shallow(<RecipeList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
