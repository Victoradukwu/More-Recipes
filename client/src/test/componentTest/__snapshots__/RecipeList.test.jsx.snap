// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeList
    deleteRecipe={[MockFunction]}
    userRecipes={
      Array [
        Object {
          "id": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th>
              Recipe Name
            </th>
            <th>
              Upvotes
            </th>
            <th>
              Downvotes
            </th>
            <th>
              Favorites
            </th>
            <th>
              views
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>,
        <tbody>
          <UserRecipe
            deleteRecipe={[MockFunction]}
            userRecipe={
              Object {
                "id": 1,
              }
            }
          />
        </tbody>,
      ],
      "className": "table tableResponsive w-auto table-hover",
      "id": "userRecipes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              Recipe Name
            </th>
            <th>
              Upvotes
            </th>
            <th>
              Downvotes
            </th>
            <th>
              Favorites
            </th>
            <th>
              views
            </th>
            <th>
              Actions
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                Recipe Name
              </th>,
              <th>
                Upvotes
              </th>,
              <th>
                Downvotes
              </th>,
              <th>
                Favorites
              </th>,
              <th>
                views
              </th>,
              <th>
                Actions
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recipe Name",
              },
              "ref": null,
              "rendered": "Recipe Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Upvotes",
              },
              "ref": null,
              "rendered": "Upvotes",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Downvotes",
              },
              "ref": null,
              "rendered": "Downvotes",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "views",
              },
              "ref": null,
              "rendered": "views",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Actions",
              },
              "ref": null,
              "rendered": "Actions",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <UserRecipe
              deleteRecipe={[MockFunction]}
              userRecipe={
                Object {
                  "id": 1,
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "deleteRecipe": [MockFunction],
              "userRecipe": Object {
                "id": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                Recipe Name
              </th>
              <th>
                Upvotes
              </th>
              <th>
                Downvotes
              </th>
              <th>
                Favorites
              </th>
              <th>
                views
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>,
          <tbody>
            <UserRecipe
              deleteRecipe={[MockFunction]}
              userRecipe={
                Object {
                  "id": 1,
                }
              }
            />
          </tbody>,
        ],
        "className": "table tableResponsive w-auto table-hover",
        "id": "userRecipes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                Recipe Name
              </th>
              <th>
                Upvotes
              </th>
              <th>
                Downvotes
              </th>
              <th>
                Favorites
              </th>
              <th>
                views
              </th>
              <th>
                Actions
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Recipe Name
                </th>,
                <th>
                  Upvotes
                </th>,
                <th>
                  Downvotes
                </th>,
                <th>
                  Favorites
                </th>,
                <th>
                  views
                </th>,
                <th>
                  Actions
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recipe Name",
                },
                "ref": null,
                "rendered": "Recipe Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Upvotes",
                },
                "ref": null,
                "rendered": "Upvotes",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Downvotes",
                },
                "ref": null,
                "rendered": "Downvotes",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "views",
                },
                "ref": null,
                "rendered": "views",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Actions",
                },
                "ref": null,
                "rendered": "Actions",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <UserRecipe
                deleteRecipe={[MockFunction]}
                userRecipe={
                  Object {
                    "id": 1,
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "deleteRecipe": [MockFunction],
                "userRecipe": Object {
                  "id": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
